paymentsheet/src/main/java/com/stripe/android/paymentsheet/ExternalPaymentMethodResultHandler.kt

```diff
@@ -0,0 +1,126 @@
+                val error = IllegalStateException(
+                    "Attempting to confirm external payment method for FlowController without an instance."
+                )
+                ErrorReporter
+                    .createFallbackInstance(context)
+                    .report(ErrorReporter.UnexpectedErrorEvent.EXTERNAL_PAYMENT_METHOD_FLOW_CONTROLLER_INSTANCE_MISSING)
+                throw error
+            } else {
+                flowControllerInstance.onExternalPaymentMethodResult(externalPaymentMethodResult.toPaymentResult())
+            }
+        }
+
+        when (integrationType) {
+            IntegrationType.PAYMENT_SHEET -> updatePaymentSheetWithResult()
+            IntegrationType.FLOW_CONTROLLER -> updateFlowControllerWithResult()
+        }
+    }
+
+    internal const val EXTRA_EXTERNAL_PAYMENT_METHOD_RESULT = "external_payment_method_result"
+    internal enum class IntegrationType { PAYMENT_SHEET, FLOW_CONTROLLER }
+    internal var integrationType: IntegrationType = IntegrationType.PAYMENT_SHEET
+}
+
+/**
+ * The result of an attempt to confirm an external payment method.
+ */
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+sealed class ExternalPaymentMethodResult : Parcelable {
+
+    internal abstract fun toPaymentResult(): PaymentResult
+
+    @Parcelize
+    internal data object Completed : ExternalPaymentMethodResult() {
+        override fun toPaymentResult(): PaymentResult = PaymentResult.Completed
+    }
+
+    @Parcelize
+    internal data object Canceled : ExternalPaymentMethodResult() {
+        override fun toPaymentResult(): PaymentResult = PaymentResult.Canceled
+    }
+
+    @Parcelize
+    internal data class Failed(private val errorMessage: String?) : ExternalPaymentMethodResult() {
+        override fun toPaymentResult(): PaymentResult = PaymentResult.Failed(
+            LocalStripeException(
+                displayMessage = errorMessage,
+                analyticsValue = "externalPaymentMethodFailure"
+            )
+        )
+    }
+
+    companion object {
+
+        /**
+         * The customer successfully completed the payment or setup.
+         */
+        @JvmStatic
+        @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+        fun completed(): ExternalPaymentMethodResult {
+            return Completed
+        }
+
+        /**
+         * The customer canceled the payment or setup attempt.
+         */
+        @JvmStatic
+        @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+        fun canceled(): ExternalPaymentMethodResult {
+            return Canceled
+        }
+
+        /**
+         * The payment or setup attempt failed.
+         *
+         * @param errorMessage Error message to display to the user on failure. If null, will display Stripe's default
+         * error message.
+         */
+        @JvmStatic
+        @JvmOverloads
+        @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+        fun failed(errorMessage: String? = null): ExternalPaymentMethodResult {
```

Comment:
Should we be more specific with this parameter name? Maybe `displayMessage` to match `CustomerAdapter.Result`?

Reply:
I _think_ the answer is update the API review doc, add a changelog entry, and notify the slack thread to let the reviewers know something changed. Since it's such a small change, it should be a quick turn around.

Reply:
awesome, thank you! Just did that